geopandas/base.py

```diff
@@ -100,6 +100,14 @@ def _generate_sindex(self):
             _sindex = sindex_cls(self.geometry)
             if not _sindex.is_empty:
                 self._sindex = _sindex
+            else:
+                warn(
+                    "Future versions of GeoPandas will allow empty spatial indexes. "
+                    "Please use `len(gdf.sindex) > 0` or `if gdf.sindex` instead of "
+                    "`if gd.sindex is not None` to check for empty spatial indexes.",
+                    FutureWarning,
```

Comment:
Can we add `stacklevel` here? (Not sure which level to point to `gdf.sindex`, so no direct suggestion.)

Reply:
It could help, but yes I'm not sure where to point to. If anyone has a concrete suggestion I'll put it in.